### ISSUE TITLE
Sales email subject "&" turns to &nbsp;

### ISSUE THREAD

**Issue by rbncha** [magento/magento2#6597]
### Preconditions

PHP 7.0.1
Mysql 5.6.31
Apache 2.4.23
Magento 2.0.7/2.1.0/2.1.1
### Steps to reproduce

1. Admin > Stores > General > General > Store Information > Store Name : change store name something having ampersand "&" like Easy&Piggy 
2. Send an order email. 
### Expected result

1. Order email subjected: Your Easy&Piggy order confirmation
### Actual result

1. Your Easy&amp;Piggy order confirmation (& is converted to &amp;)
2. 
   ![image](https://cloud.githubusercontent.com/assets/767121/18475391/f31e09ca-79e5-11e6-9595-68789de4879e.png)

**Comment by veloraven** [comment]
@rbncha thank you for your report.
Could you please add some formatting to the last line? As I see "& is converted to &" instead of "& is converted to &amp;" and it's confusing.
You can just add "\" sign before &amp;

**Comment by quienti** [comment]
Probably same problem, but with mail subject where ' is &#039;
Add '\' do not works.
On Magento 2.1.1

![sans titre 1](https://cloud.githubusercontent.com/assets/7326954/18677307/adccbd4e-7f58-11e6-8652-b8a7cc25e4ad.jpg)

**Comment by olysenko** [comment]
Hi, 
internal ticket is created MAGETWO-58663

**Comment by calvintwr** [comment]
any updates on this issue? i see that this issue has no assignment.

**Comment by PieterCappelle** [comment]
Any update on this one?

**Comment by PieterCappelle** [comment]
Quick fix is in `di.xml`

```<?xml version="1.0"?>
<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:ObjectManager/etc/config.xsd">
    <type name="Magento\Email\Model\Template">
        <plugin name="mail_subject" type="VENDOR\MODULE\Plugin\Template" sortOrder="100" />
    </type>
</config>
```

In Template.php
```<?php
namespace VENDOR\MODULE\Plugin;

class Template
{
    public function afterGetSubject(
        \Magento\Email\Model\Template $subject,
        $result
    ) {
        return htmlspecialchars_decode((string)$result, ENT_QUOTES);
    }
}
```

**Comment by magento-engcom-team** [comment]
@rbncha, thank you for your report.
We were not able to reproduce this issue by following the steps you provided. If you'd like to update it, please reopen the issue.
We tested the issue on 2.3.0-dev, 2.2.0, 2.1.9

**Comment by korostii** [comment]
Hi @magento-engcom-team

> We tested the issue on 2.3.0-dev, 2.2.0, 2.1.9

That can't be right. As mentioned on the linked PR #9476 , here's the fix: e89d09d2885b51c49c223ff795041fd6aab5fafc
And it seems to be present on 2.2 branch only.
